op-node/node/safedb/safedb.go

```diff
@@ -0,0 +1,192 @@
+package safedb
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math"
+	"slices"
+	"sync"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var (
+	ErrNotFound     = errors.New("not found")
+	ErrInvalidEntry = errors.New("invalid db entry")
+)
+
+const (
+	// Keys are prefixed with a constant byte to allow us to differentiate different "columns" within the data
+	keyPrefixSafeByL1BlockNum byte = 0
+)
+
+var (
+	SafeByL1BlockNumKey = uint64Key{prefix: keyPrefixSafeByL1BlockNum}
```

Comment:
I see how this works & it's fine, but it's a new pattern to me.

Reply:
I'm certainly open to suggestions for other patterns of handling column prefixed keys like this but it was pretty messy originally just trying to use functions and the uint64Key struct really helped clean it up.